src/Data/Array/Mutable/Linear.hs

```diff
@@ -81,6 +81,13 @@ alloc size x f
   | size > 0 = f (Array size (Unsafe.newMutArr size x))
   | otherwise = error $ "Trying to allocate an array of size " ++ show size
 
+-- | Allocate a constant array given a size and an initial value,
+-- using another array as a uniqueness proof.
+allocBeside :: Int -> a -> Array b #-> (Array b, Array a)
```

Comment:
```suggestion
-- using another value as a uniqueness proof.
allocBeside :: Int -> a -> b #-> (b, Array a)
```
Does it matter if the source of uniqueness is an Array or something else?

Reply:
I'm finding this aspect of dealing with linear types rather slippery. It may be just me, though.

Reply:
> I'm finding this aspect of dealing with linear types rather slippery. It may be just me, though.

I do agree. This `allocBeside` function is pretty arbitrary here, as @aspiwack mentioned there is a lot of possible other types that would work just as well (`Vector.Mutable.Linear` and our other collections is the first that came to my mind), so there's another "m*n" problem here. Another arbitrary decision is that only `Array` provides this function, while its likes with the similar properties doesn't.

However, I do think we should use #130 for this discussion and this function could be refactored later.